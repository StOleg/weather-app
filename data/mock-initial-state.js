export const initialState = {
    searchText: '',
    loading: false,
    error: false,
    city: {},
    list: []
};