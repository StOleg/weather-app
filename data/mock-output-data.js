export const outputData = {
    city: {
        coord: { lat: 49.842, lon: 24.0316 },
        country: "UA",
        id: 702550,
        name: "Lviv"
    },
    dailyList: [
        {
            dateShort: "7-1-2019",
            dayName: "Monday",
            hourlyDetails: [{
                clouds: 68,
                hours: "21:00",
                humidity: 82,
                pressure: 1004.95,
                temp: "-14.21°C",
                timeId: 1546894800,
                weatherDescription: "light snow",
                weatherIconUrl: "http://openweathermap.org/img/w/13n.png",
                weatherMain: "Snow",
                windDirection: 231.506,
                windSpeed: 1.41,
            }]

        },
        {
            dateShort: "8-1-2019",
            dayName: "Tuesday",
            hourlyDetails: [
                {
                    clouds: 8,
                    hours: "0:00",
                    humidity: 84,
                    pressure: 1002.2,
                    temp: "-14.38°C",
                    timeId: 1546905600,
                    weatherDescription: "light snow",
                    weatherIconUrl: "http://openweathermap.org/img/w/13n.png",
                    weatherMain: "Snow",
                    windDirection: 168,
                    windSpeed: 2.27,
                },
                {
                    clouds: 0,
                    hours: "3:00",
                    humidity: 77,
                    pressure: 999.09,
                    temp: "-15.23°C",
                    timeId: 1546916400,
                    weatherDescription: "clear sky",
                    weatherIconUrl: "http://openweathermap.org/img/w/01n.png",
                    weatherMain: "Clear",
                    windDirection: 165,
                    windSpeed: 3.9
                }
            ]
        }
    ]
};