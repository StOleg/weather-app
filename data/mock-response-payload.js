export const responcePayload = {
    city: {
        country: "UA",
        id: 702550,
        name: "Lviv"
    },
    list: [
        {
            clouds: {
                all: 68
            },
            main: {
                humidity: 82,
                pressure: 1004.95
            }

        },
        {
            clouds: {
                all: 50
            },
            main: {
                humidity: 70,
                pressure: 1004.95
            }

        }
    ]
};