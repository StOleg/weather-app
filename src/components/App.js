import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { simpleAction } from '../actions/simpleAction';

export class App extends Component {
	constructor(props) {
		super(props);
		this.handleClick = this.handleClick.bind(this);
	}

	handleClick() {
		this.props.simpleAction();
	}

	render() {
		return (
			<div className='App'>
				<h1>Hello, React!!</h1>
				<button onClick={this.handleClick}>Test redux action</button>
				<pre>
					{JSON.stringify(this.props)}
				</pre>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	...state
});

const mapDispatchToProps = {
	simpleAction
};

App.propTypes = {
	simpleAction: PropTypes.func.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(App);