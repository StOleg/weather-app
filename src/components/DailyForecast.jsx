import React from 'react';
import PropTypes from 'prop-types';

const DailyForecast = (props) => {
	return (
		<div>
			<div className='cityName'>
				{props.city.name}
			</div>
			{props.dailyList.map((day) =>
				<div key={day.dateShort} className='dayWeather'>
					<div className='dayDetails'>
						<span>{day.dayName}</span>
						<span>{day.dateShort}</span>
					</div>
					{day.hourlyDetails.map((details) =>
						<div key={details.timeId} className='hourlyDetails'>
							<span>{details.hours}</span>
							<span>{details.weatherMain}</span>
							<span>{details.temp}</span>
							<span><img className='rounded-circle' src={details.weatherIconUrl} alt={details.weatherDescription} height='50' width='50' /></span>
							<span>{details.weatherDescription}</span>
							<span>{details.humidity}</span>
							<span>{details.pressure}</span>
							<span>{details.clouds}</span>
							<span>{details.windSpeed}</span>
							<span>{details.windDirection}</span>
						</div>
					)}
				</div>
			)}
		</div>
	);
};

DailyForecast.propTypes = {
	city: PropTypes.object.isRequired,
	dailyList: PropTypes.array.isRequired
};

export default DailyForecast;