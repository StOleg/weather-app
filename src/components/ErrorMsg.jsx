import React from 'react';

export const ErrorMsg = () => {
    return (
        <div className="error_msg">
            <p>Error, try again</p>
        </div>
    );
};

export default ErrorMsg;