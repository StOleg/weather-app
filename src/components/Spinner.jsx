import React from 'react';

export const Spinner = () => {
    return (
        <div className="spinner_container">
            <div className="spinner"></div>
        </div>
    );
};

export default Spinner;