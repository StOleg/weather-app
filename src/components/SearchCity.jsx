import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Spinner from './Spinner';
import ErrorMsg from './ErrorMsg';
import DailyForecast from './DailyForecast';

export class SearchCity extends Component {
	constructor(props) {
		super(props);
		this.handleSearch = this.handleSearch.bind(this);
		this.inputSearch = React.createRef();
	}

	static propTypes = {
		fetchCity: PropTypes.func.isRequired,
		error: PropTypes.bool.isRequired,
		loading: PropTypes.bool.isRequired,
		city: PropTypes.object.isRequired,
		dailyList: PropTypes.array.isRequired
	}

	handleSearch() {
		const searchTextInput = this.inputSearch.current.value;
		this.props.fetchCity(searchTextInput);
	}

	render() {
		const { loading, error, city, dailyList } = this.props;
		return (
			<div className='app'>
				<header>
					<h1>Weather in your city</h1>
				</header>
				<div className='input-group input-group-lg'>
					<input
						type='text'
						required='required'
						ref={this.inputSearch}
						placeholder='Ex. Lviv'
					/>
					<input
						type='button'
						value='Search'
						className='btn btn-dark'
						onClick={this.handleSearch}
					/>
				</div>
				{loading
					? <Spinner />
					: error
						? <ErrorMsg />
						: <DailyForecast city={city} dailyList={dailyList} />
				}
			</div>
		);
	}
}

export default SearchCity;