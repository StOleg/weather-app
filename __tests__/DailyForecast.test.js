import React from 'react';
import TestRenderer from 'react-test-renderer';
import { DailyForecast } from '../src/components/DailyForecast';
import objectAssignDeep from 'object-assign-deep';
import { outputData } from '../data/mock-output-data';

describe('>>> DailyForecast Component', () => {
    let data;
    it('renders as expected', () => {
        data = objectAssignDeep(outputData);
        const tree = TestRenderer.create(<DailyForecast city={data.city} dailyList={data.dailyList} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});