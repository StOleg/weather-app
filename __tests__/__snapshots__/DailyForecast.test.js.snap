// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> DailyForecast Component renders as expected 1`] = `
<div>
  <div
    className="cityName"
  >
    Lviv
  </div>
  <div
    className="dayWeather"
  >
    <div
      className="dayDetails"
    >
      <span>
        Monday
      </span>
      <span>
        7-1-2019
      </span>
    </div>
    <div
      className="hourlyDetails"
    >
      <span>
        21:00
      </span>
      <span>
        Snow
      </span>
      <span>
        -14.21°C
      </span>
      <span>
        <img
          alt="light snow"
          className="rounded-circle"
          height="50"
          src="http://openweathermap.org/img/w/13n.png"
          width="50"
        />
      </span>
      <span>
        light snow
      </span>
      <span>
        82
      </span>
      <span>
        1004.95
      </span>
      <span>
        68
      </span>
      <span>
        1.41
      </span>
      <span>
        231.506
      </span>
    </div>
  </div>
  <div
    className="dayWeather"
  >
    <div
      className="dayDetails"
    >
      <span>
        Tuesday
      </span>
      <span>
        8-1-2019
      </span>
    </div>
    <div
      className="hourlyDetails"
    >
      <span>
        0:00
      </span>
      <span>
        Snow
      </span>
      <span>
        -14.38°C
      </span>
      <span>
        <img
          alt="light snow"
          className="rounded-circle"
          height="50"
          src="http://openweathermap.org/img/w/13n.png"
          width="50"
        />
      </span>
      <span>
        light snow
      </span>
      <span>
        84
      </span>
      <span>
        1002.2
      </span>
      <span>
        8
      </span>
      <span>
        2.27
      </span>
      <span>
        168
      </span>
    </div>
    <div
      className="hourlyDetails"
    >
      <span>
        3:00
      </span>
      <span>
        Clear
      </span>
      <span>
        -15.23°C
      </span>
      <span>
        <img
          alt="clear sky"
          className="rounded-circle"
          height="50"
          src="http://openweathermap.org/img/w/01n.png"
          width="50"
        />
      </span>
      <span>
        clear sky
      </span>
      <span>
        77
      </span>
      <span>
        999.09
      </span>
      <span>
        0
      </span>
      <span>
        3.9
      </span>
      <span>
        165
      </span>
    </div>
  </div>
</div>
`;
